cookbook/upgrading.rst

```diff
@@ -0,0 +1,127 @@
+How to Upgrade your Symfony Project
+===================================
+
+So a new Symfony release has come out and you want to upgrade, great! Fortunately,
+because Symfony protects backwards-compatibility very closely, this *should*
+be quite easy.
+
+Upgrading a Patch Version (e.g. 2.6.0 to 2.6.1)
+-----------------------------------------------
+
+If you're upgrading and only the patch version (the last number) is changing,
+then it's *really* easy:
+
+.. code-block:: bash
+
+    $ composer update symfony/symfony
+
+That's it! You should not encounter any backwards-compatability breaks or
+need to change anything else in your code.
+
+You may also want to upgrade the rest of your libraries. If you've done a
+good job with your version constraints in ``composer.json``, you can do this
+safely by running:
+
+.. code-block:: bash
+
+    $ composer update symfony/symfony
+
+But beware. If you have some bad version constraints in your ``composer.json``,
+(e.g. ``dev-master``), then this could upgrade some non-Symfony libraries
+to new versions that contain backwards-compability changes.
+
+Upgrading a Minor Version (e.g. 2.5.3 to 2.6.0)
+-----------------------------------------------
+
+If you're upgrading a minor version (where the middle number changes), then
+you should also *not* encounter significant backwards compability changes.
+For details, see our :doc:`/contributing/code/bc`.
+
+However, some backwards-compability breaks *are* possible, and you'll learn
+in a second how to prepare for them.
+
+There are two steps to upgrading:
+
+1. :ref:`upgrade-minor-symfony-composer`;
+2. :ref:`upgrade-minor-symfony-code`, which includes instructions for each version.
+
+.. _`upgrade-minor-symfony-composer`:
+
+Update the Symfony Library
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, you need to update Symfony by modifying your ``composer.json`` to
+use the new version:
+
+.. code-block:: json
+
+    {
+        "...": "...",
+
+        "require": {
+            "php": ">=5.3.3",
+            "symfony/symfony": "~2.6.0",
+            "...": "... no changes to anything else..."
+        },
+        "...": "...",
+    }
+
+Next, update the same as before:
+
+.. code-block:: bash
+
+    $ composer update symfony/symfony
+
+Updating a minor version like this should *not* cause any dependency issues,
+though it's always possible that an outside library or bundle you're using
+didn't support this new version of Symfony at the version you have of that
+library.  In that case, consult the library: you may need to modify its version
+in ``composer.json`` and run a full ``composer update``. 
```

Comment:
Actually, nevermind. I think that using `require` isn't the best idea for Symfony, because we tell people to use `2.6.*`-style requirements, instead of `~2.6`, I guess to be a bit more conservative. Using `require` would - I believe - give us the tilde version. I did sort of re-word this section though. I think we should actually tell people to use `composer update` to upgrade their libraries. I think if they actually want a full new version of some non-Symfony library that's beyond what they originally put in their composer.json file, then they're on their own to do that (and this would be a common thing to do with Composer). Also, assuming more and more people use `composer require` to initially add a library, people will have constraints like `~1.2` more and more commonly, which means "update" will allow them patch and minor versions (and it make sense for them to knowingly opt-in if they want to change the major version).